core/src/main/java/io/questdb/griffin/SqlParser.java

```diff
@@ -1737,18 +1752,29 @@ private void parseSelectClause(GenericLexer lexer, QueryModel model) throws SqlE
                 }
                 tok = optTok(lexer);
             } else {
-                alias = createColumnAlias(expr, model);
+                alias = null;
+                col.setAlias(null);
+                columns.add(col);
             }
-            if (alias.length() == 0) {
-                throw err(lexer, null, "column alias cannot be a blank string");
+
+            if (alias != null) {
```

Comment:
nit: better to concat two ifs which check alias for null

```java
            if (alias != null) {
                if (alias.length() == 0) {
                    throw err(lexer, null, "column alias cannot be a blank string");
                }
                col.setAlias(alias);
                model.addBottomUpColumn(colPosition, col, false);
            } else { //only add to bottom up columns / bottom up column names which are responsible for ordering, the others will be updated later
                model.addFieldForNullAlias(col);
            }
```

Reply:
Fixed in https://github.com/questdb/questdb/pull/3644/commits/b06bfa942018ee4f648f824a8e90485f624eb078